ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -1989,3 +1993,30 @@ def get_logsize(host, logfile):
     """ get current logsize"""
     logsize = len(host.get_file_contents(logfile))
     return logsize
+
+
+def run_ssh_command_as_user(host, username, password, cmd, raiseonerr=True):
+    """
```

Comment:
Another method to get a TGT is to call:
```
cmd = ['su', '-l', username, '-c', cmd]
tasks.run_command_as_user(host, username, cmd, stdin_text=password + '\n')
```
as this will run `su -l username -c 'su -l username -c cmd'`

Reply:
Of course, there are several ways to obtain a TGT.
But I think this could be useful since it exercises a different PAM.

Reply:
@stanislavlevin well, in this particular case I don't see why we should be using ssh connection to the `nfs_autofs` host, there is simply no need to go that way.

If you want to add a test that tests SSH login with password, please create a new test. However, for NFS test I don't think it is warrant to add a new way of accessing the host.

Note that actual access to the host from test controller might be done differently and you simply might not be able to reach it via ssh in some environments. I'd prefer to keep this abstraction to a paramiko/test environment setup.

Reply:
@abbra, I see how it is made in `ipatests/test_integration/test_smb.py` and I can't say that it fits my needs.

```
    @classmethod                                                                
    def install(cls, mh):                                                       
        if cls.domain_level is not None:                                        
            domain_level = cls.domain_level                                     
        else:                                                                   
            domain_level = cls.master.config.domain_level                       
        tasks.install_topo(cls.topology,                                        
                           cls.master, cls.replicas,                            
                           cls.clients, domain_level,                           
                           clients_extra_args=('--mkhomedir',))
```
Why:
1) The proposed test checks the auto mounting user home directory located on NFS share. The mixing of the ways to provide a home directory for a user is not that I want to check here. In other words, I don't want something else touches the home dir, even if the autofs hides an original one.
2) The proposed test checks a real user's interaction with his environment and in particular,  his home dir. In my opinion, this is a cleaner way rather than `su ... su ...`.

So, `--mkhomedir` and `su` are just another one workaround.

Note: without local `mkhomedir` `su` fails (at least on ALT):
```
INFO     ipatests.pytest_ipa.integration.host.Host.client3.ParamikoTransport:transport.py:318 RUN ['su', '-l', 'athena', '-c', 'su -l athena -c klist']
DEBUG    ipatests.pytest_ipa.integration.host.Host.client3.cmd65:transport.py:513 RUN ['su', '-l', 'athena', '-c', 'su -l athena -c klist']
DEBUG    ipatests.pytest_ipa.integration.host.Host.client3.cmd65:transport.py:558 /home/athena not available; exiting
DEBUG    ipatests.pytest_ipa.integration.host.Host.client3.cmd65:transport.py:217 Exit code: 1
ERROR    ipatests.pytest_ipa.integration.host.Host.client3.cmd65:host.py:74 stderr: /home/athena not available; exiting
```


> Note that actual access to the host from test controller might be done differently and you simply might not be able to reach it via ssh in some environments. I'd prefer to keep this abstraction to a paramiko/test environment setup.

Really? not SSH? Could I see such examples?
What is the kind of restrictions?
I change user `root -> test_user` and authentication type (pub key -> interactive password).

All of this is not something I'm going to dispute, just my view.

Reply:
For `test_smb` tests, no automount test is being done, so autocreating home directory is OK there. Samba needs the home directory to exist when connecting as the user and autocreation cannot happen on connection to Samba itself, it needs to be created in advance.

As for 'no SSH', we should, in general, not assume particular details about the transport used in test to access test hosts. Today it is native ssh or Paramiko implementation, tomorrow it might be another method. Think of Kubernetes, for example -- if these tests would at some point be running in k8s environment, getting access to the specific instance would be using a different method than ssh. I don't want us to re-do all the tests due to this change.

Reply:
I replaced my hack for ssh with `su`.